src/services/useMenuItemsStack/index.ts

```diff
@@ -0,0 +1,45 @@
+import { ColumnItem } from '@/components/general/ColumnItem'
+import { ref, Ref, MaybeRefOrGetter, toValue, watch } from 'vue'
+
+export function useMenuItemsStack(
+  menuItems: MaybeRefOrGetter<ColumnItem[]>,
+  activeItem: MaybeRefOrGetter<string>,
+): Ref<ColumnItem[]> {
+  const stack = ref<ColumnItem[]>([])
+
+  watch(menuItems, updateStack, { immediate: true })
+  watch(() => activeItem, updateStack)
+  function updateStack() {
+    const _menuItems = toValue(menuItems)
+    const _activeItem = toValue(activeItem)
+
+    if (_activeItem !== '' && _menuItems.length > 0) {
+      const root: ColumnItem = {
+        id: 'rootNode',
+        name: '',
+        children: _menuItems,
+      }
+      const s = [root]
```

Comment:
Could consider:
```suggestion
      stack.value = recursiveFind(root, _activeItem)
    }
  }
  return stack
}

function recursiveFind(item: ColumnItem, id: string): ColumnItem[] {
  if (item.id === id) return [item]

  for (const child of item.children ?? []) {
    const res = recursiveFind(child, id)
    if (res.length > 0) {
      return [item, ...res]
    }
  }

  return []
}

```

Reply:
Behavior in the column menu is slightly different, so I won't adopt this.